iphone/Classes/TiUIiOSApplicationShortcutsProxy.m

```diff
@@ -173,19 +204,44 @@ -(void)addShortcutItem:(id)args
 
 -(UIApplicationShortcutIcon*)findIcon:(id)value
 {
+    if ([value isKindOfClass:[UIApplicationShortcutIcon class]]) {
+        return (UIApplicationShortcutIcon*)value;
+    }
+    
     if ([value isKindOfClass:[NSNumber class]]) {
         NSInteger iconIndex = [value integerValue];
         return [UIApplicationShortcutIcon iconWithType:iconIndex];
     }
     
     if ([value isKindOfClass:[NSString class]]) {
-        return [UIApplicationShortcutIcon iconWithTemplateImageName:(NSString*)value];
+        return [UIApplicationShortcutIcon iconWithTemplateImageName:[self urlInAssetCatalog:value]];
     }
     
-    NSLog(@"[ERROR] Invalid icon selection provided, defaulting to SHORTCUT_ICON_TYPE_COMPOSE");
+    NSLog(@"[ERROR] Invalid icon provided, defaulting to SHORTCUT_ICON_TYPE_COMPOSE.");
     return UIApplicationShortcutIconTypeCompose;
 }
 
+-(NSString*)urlInAssetCatalog:(NSString*)url
```

Comment:
Similar stuff happens here:
https://github.com/appcelerator/titanium_mobile/pull/7248/files

Maybe should be in one util class?